src/Tribe/Views/V2/Widgets/Compatibility.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Compatibility for Advanced List Widgets and List Widget.
+ * In V2 Advanced List Widgets are merged with List Widgets, it is reversed with V1.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Pro\Views\V2\Widgets
+ */
+
+namespace Tribe\Events\Views\V2\Widgets;
+
+/**
+ * Class Compatibility
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Widgets
+ */
+class Compatibility {
+
+	/**
+	 * The default primary list widget id base string.
+	 *
+	 * @var string
+	 */
+	protected $primary_id_base = 'tribe-events-list-widget';
+
+	/**
+	 * The default alternative list widget id base string.
+	 *
+	 * @var string
+	 */
+	protected $alternative_id_base = 'tribe-events-adv-list-widget';
+
+	/**
+	 * Adds the filters for V2 Widgets
+	 *
+	 * @since TBD
+	 */
+	public function hooks() {
+		add_filter( 'tribe_plugins_loaded', [ $this, 'switch_to_v1_compatibility' ] );
+		add_filter( 'option_sidebars_widgets', [ $this, 'remap_list_widget_id_bases' ] );
+		add_filter( 'option_widget_tribe-events-list-widget', [ $this, 'merge_list_widget_options' ] );
+	}
+
+	/**
+	 * Switches the primary and alternative id base when v1 is active
+	 * or v2 widgets are disabled, this enables support for upgrading from
+	 * v1 free list widget to the v1 Pro advanced list widget.
+	 *
+	 * @since TBD
+	 *
+	 */
+	public function switch_to_v1_compatibility() {
```

Comment:
:no_entry_sign: Additional blank lines found at end of doc comment

<sub>hash: be297244ace0dc2fd73ffa29408cbace</sub>